src/app/notes/notes.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { NoteService } from '../services/note.service';
+import { Note } from '../models/note';
+import { Subject, tap } from 'rxjs';
+
+@Component({
+  selector: 'app-notes',
+  templateUrl: './notes.component.html',
+  styleUrls: ['./notes.component.scss']
+})
+export class NotesComponent implements OnInit {
+  displayedColumns: string[] = ['id', 'name', 'note', 'category', 'delete'];
+  noteList: Note[] = [];
+  currentNote!: Note;
+
+  constructor(public noteService: NoteService) {}
+
+  ngOnInit() {
+    this.getNotes();
+  }
+
+  updateDetails(note: Note) {
+    if (note.id !== 0) {
+      this.noteService.updateNoteDetails(note)
+        .pipe(
+          tap(n => this.noteList = this.noteList.map(curNote => curNote.id === note.id ? n : curNote)),
```

Comment:
Instead of creating a copy of the notes for your component, you can just subscribe and then reference the service's copy of the note list. That way your note state has a single source of truth instead of multiple copies floating around that you have to keep in sync.